lib/mongo/connection/node.rb

```diff
@@ -73,6 +73,16 @@ def connect
       end
     end
 
+    def socket
+      if @socket && @socket.pid != Process.pid
```

Comment:
The `pid` property is assigned when a socket is created:

https://github.com/mongodb/mongo-ruby-driver/blob/1.12.0/lib/mongo/connection/socket/ssl_socket.rb#L28
https://github.com/mongodb/mongo-ruby-driver/blob/1.12.0/lib/mongo/connection/socket/tcp_socket.rb#L27
https://github.com/mongodb/mongo-ruby-driver/blob/1.12.0/lib/mongo/connection/socket/unix_socket.rb#L28

This is the same logic being used already in the connection pool code:

https://github.com/mongodb/mongo-ruby-driver/blob/1.12.0/lib/mongo/connection/pool.rb#L224